runtime/src/mm/cpp/ThreadData.hpp

```diff
@@ -17,14 +18,25 @@ namespace mm {
 // Pin it in memory to prevent accidental copying.
 class ThreadData final : private Pinned {
 public:
+    enum class State {
+        kRunning,
+        kWaitGC,
+    };
```

Comment:
The resulting thread state machinery is going to be different anyway. Maybe don't expose and even implement it at the moment, and make a predicate like `isWaitingForGC` instead?

Reply:
Sure, replaced with a predicate.